docs/book/src/design/wallet-tee.md

```diff
@@ -0,0 +1,25 @@
+# Wallet TEE
+
+The Wallet TEE consists of a trusted SGX enclave, alongside an untrusted host service.
+
+## SGX enclave
+
+The Wallet SGX enclave contains the system's [Trusted Computing Base (TCB)][TCB], and is responsible for all security-sensitive wallet operations:
+
+- Generating and managing account keypairs
+- Performing privileged operations such as transaction signing
+
+Once generated, secret keys and other sensitive data never leaves the enclave unencrypted. All persistent storage is cryptographically sealed and authenticated, and all trusted communication with the enclave occurs over cryptographically attested and mutually-authenticated channels.
+
+[TCB]: https://en.wikipedia.org/wiki/Trusted_computing_base
+
+## Host service
+
+The SGX enclave is embedded in an untrusted host service, which facilitates communication between the enclave and the outside world.
+
+* Remote attestation, which allows the Connector app to remotely verify the identity and integrity of the enclave, and exchange keys to set up a trusted communication channel
```

Comment:
At this point in the documentation, it has not yet been made clear what the *Connector app* is.  I would suggest one of the following:

1. Change the order of the chapters so that the "Connector app" chapter appears before this one
2. Keep the order the same but refer to the Connector simply as the "client application"

Reply:
Yeah; we should probably give more context for this in the parent Wallet chapter.

I don't think changing the order addresses the dependency problem: describing the Connector app client component also doesn't make sense without describing the TEE server component, so either order will require a sibling reference. (I weakly lean towards keeping the TEE server first, just because it's more architecturally foundational.)

For the naming, I lean towards "Connector app" mainly for consistency with our other documentation and material. It may not be the final name, but we should probably stick to it wherever we can, for now, if only so that it's easier to find and update it later.

Reply:
I agree, this can de done in a simple fashion in the parent chapter.  I did not consider that possibiliity but it is certainly much more elegant! :+1:

Reply:
Should we change anything in the parent chapter to resolve this, or is it okay as-is, for now?

Reply:
I think as it stands, it is sufficiently minor so as to address in a future PR.  I'm pulling it out into a new issue for bookkeeping.